elfio/elfio_relocation.hpp

```diff
@@ -319,14 +319,17 @@ template <class S> class relocation_section_accessor_template
 
     //------------------------------------------------------------------------------
     template <class T>
-    void generic_get_entry_rel( Elf_Xword   index,
+    bool generic_get_entry_rel( Elf_Xword   index,
                                 Elf64_Addr& offset,
                                 Elf_Word&   symbol,
                                 unsigned&   type,
                                 Elf_Sxword& addend ) const
     {
         const endianess_convertor& convertor = elf_file.get_convertor();
 
+        if (relocation_section->get_entry_size() < sizeof( T ) ) {
```

Comment:
For my understanding, why is this not a == check?

From the PR description, it sounds like validating for 'expected' type would be an == check (but probably this check is more about validity of memory access?)

Reply:
I believe the convention is that types in the ELF file are permitted to be larger (to permit future additions). The code onl requires that at least that many fields are valid.

Reply:
As an existing example, see:
https://github.com/serge1/ELFIO/blob/main/elfio/elfio_dynamic.hpp#L54